### ISSUE TITLE
Upgrading to 8.0.0-rc.2.9511 breaks the app. Loses references to MauiUIApplicationDelegate

### ISSUE THREAD

**Issue by Forgind** [dotnet/maui#18739]
---
**Issue moved from dotnet/sdk#36667**
- Please respond to @fady-intelligencex.

---

_From @fady-intelligencex on Saturday, November 4, 2023 7:01:32 PM_

### Description

Once I upgraded to 8.0.0-rc.2.9511, the app stopped building with the Error CS0246: The type or namespace name 'MauiUIApplicationDelegate' could not be found (are you missing a using directive or an assembly reference?) (CS0246). To validate, I created a new project. It works fine with the a previous version, but once I upgrade, it has the same problem.

### Steps to Reproduce

1. Create new project in maui .net 8.0
2. Run the project
3. Update Microsoft.Maui.Controls to 8.0.0-rc-2.9511
4. Attempt to run project
5. see error in the build output.

under Platforms/iOS/ in the file AppDelegate.cs 

using Foundation;

namespace newtest4;

[Register("AppDelegate")]
public class AppDelegate : MauiUIApplicationDelegate
{
	protected override MauiApp CreateMauiApp() => MauiProgram.CreateMauiApp();
}

It cannot find: MauiUIApplicationDelegate

I shared a video link where I demo it happening on a new project.

### Link to public reproduction project repository

https://youtu.be/aBUKjdeMXz8

### Version with bug

8.0.0-rc.2.9511

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

8.0.0-rc.1.9171

### Affected platforms

iOS

### Did you find any workaround?

N/A

### Relevant log output

```shell
Projects/newtest4/newtest4/Platforms/iOS/AppDelegate.cs(28,28): Error CS0246: The type or namespace name 'MauiUIApplicationDelegate' could not be found (are you missing a using directive or an assembly reference?) (CS0246) (newtest4)
```

**Comment by Forgind** [comment]
---
**Issue moved from dotnet/sdk#36667**
- Please respond to @msftbot[bot].

---

_From @msftbot[bot] on Monday, November 6, 2023 3:05:31 PM_

Hi @fady-intelligencex. We have added the "s/needs-info" label to this issue, which indicates that we have an open question for you before we can take further action. This issue will be closed automatically in 7 days if we do not hear back from you by then - please feel free to re-open it if you come back to this issue after that time.

**Comment by Forgind** [comment]
---
**Issue moved from dotnet/sdk#36667**
- Please respond to @DDHSchmidt.

---

_From @DDHSchmidt on Monday, November 6, 2023 3:12:54 PM_

> I had the same issue actually, so I have done the following steps to fix it:
> 
> 1. run `dotnet workload update`
> 2. It should update your workloads to the latest workloads that the 8.0.0-RC.2.9511 uses.
> 3. Close VS and reopen it again then you should see your issue is fixed.
> 
> I actually use the latest SDK RC2 8 Core (8.0.100-rc.2.23502.2) https://dotnet.microsoft.com/en-us/download/dotnet/8.0

I have the same problem as the original poster and this did **not** fix my problem.

dotnet --list-runtimes contains `Microsoft.NETCore.App 8.0.0-rc.2.23479.6 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]`

dotnet --list-sdks contains `8.0.100-rc.2.23502.2 [/usr/local/share/dotnet/sdk]`

dotnet workload list displays:
```
Installed Workload Id      Manifest Version                     Installation Source
-----------------------------------------------------------------------------------
maui-ios                   8.0.0-rc.1.9171/8.0.100-rc.1         SDK 8.0.100-rc.2   
ios                        16.4.8825-net8-rc1/8.0.100-rc.1      SDK 8.0.100-rc.2   
maui                       8.0.0-rc.1.9171/8.0.100-rc.1         SDK 8.0.100-rc.2
```

This is *after* I ran `sudo dotnet workload update --include-previews`

I'm using Visual Studio for Mac 17.6.6 with XCode 15.0.1

If you need more advice, don't hesitate too ask **quickly**. I'm deeply concerned about this dependency mixup in an addendum to a Release Candidate this close before GA :(

**Comment by Forgind** [comment]
---
**Issue moved from dotnet/sdk#36667**
- Please respond to @DDHSchmidt.

---

_From @DDHSchmidt on Monday, November 6, 2023 3:31:24 PM_

Discord advised me to provide a binlog.
Here is the 
result of `dotnet build VanillaMauiApp.sln /bl:msbuild.binlog` : [msbuild.binlog.zip](https://github.com/dotnet/sdk/files/13268888/msbuild.binlog.zip)
I also provide the sources of the vanilla app I used to trigger the faulty result: 
[VanillaMauiApp.zip](https://github.com/dotnet/sdk/files/13268906/VanillaMauiApp.zip)

**Comment by Forgind** [comment]
---
**Issue moved from dotnet/sdk#36667**
- Please respond to @AndreasBreuninger.

---

_From @AndreasBreuninger on Thursday, November 9, 2023 9:35:50 AM_

> > I had the same issue actually, so I have done the following steps to fix it:
> > 
> > 1. run `dotnet workload update`
> > 2. It should update your workloads to the latest workloads that the 8.0.0-RC.2.9511 uses.
> > 3. Close VS and reopen it again then you should see your issue is fixed.
> > 
> > I actually use the latest SDK RC2 8 Core (8.0.100-rc.2.23502.2) https://dotnet.microsoft.com/en-us/download/dotnet/8.0
> 
> I have the same problem as the original poster and this did **not** fix my problem.
> 
> dotnet --list-runtimes contains `Microsoft.NETCore.App 8.0.0-rc.2.23479.6 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]`
> 
> dotnet --list-sdks contains `8.0.100-rc.2.23502.2 [/usr/local/share/dotnet/sdk]`
> 
> dotnet workload list displays:
> 
> ```
> Installed Workload Id      Manifest Version                     Installation Source
> -----------------------------------------------------------------------------------
> maui-ios                   8.0.0-rc.1.9171/8.0.100-rc.1         SDK 8.0.100-rc.2   
> ios                        16.4.8825-net8-rc1/8.0.100-rc.1      SDK 8.0.100-rc.2   
> maui                       8.0.0-rc.1.9171/8.0.100-rc.1         SDK 8.0.100-rc.2
> ```
> 
> This is _after_ I ran `sudo dotnet workload update --include-previews`
> 
> I'm using Visual Studio for Mac 17.6.6 with XCode 15.0.1
> 
> If you need more advice, don't hesitate too ask **quickly**. I'm deeply concerned about this dependency mixup in an addendum to a Release Candidate this close before GA :(

Had the same issue. For me it was fixed with these steps:

1. dotnet workload update 
2. Close Visual Studio
3. Delete bin and obj folder

**Comment by Forgind** [comment]
---
**Issue moved from dotnet/sdk#36667**
- Please respond to @anpin.

---

_From @anpin on Tuesday, November 14, 2023 9:45:58 AM_

I had to run `sudo dotnet workload update --include-previews` from a place without `nuget.config` with private source or it was showing me this
![image](https://github.com/dotnet/sdk/assets/6060545/a29dd941-6f79-4d5c-b75a-f817a4b5489d)

**Comment by Redth** [comment]
There was a change between previews to the iOS workload which required an update to that workload in order to use the newer MAUI package.  This should be resolved with the GA release.

Typically we will avoid such changes between service releases without coordinating the updated workload set for MAUI as well, but unfortunately for preview releases this isn't always possible.

**Comment by wallym** [comment]
I am running on the .NET 8 GA.  I have the released VS.net 2022.  I have updated my workloads.  I get this error.

**Comment by MLB-BIOMETRIC** [comment]
I have this issue as well.

Just installed .NET 8 GA with the workloads below, but still issues.

```
Installed Workload Id      Manifest Version      Installation Source
--------------------------------------------------------------------
maui                       8.0.3/8.0.100         SDK 8.0.100        
```

```
 The type or namespace name 'MauiUIApplicationDelegate' could not be found (are you missing a using directive or an assembly reference?)
 ```

**Comment by michaelstonis** [comment]
I don't believe this will help for all situations, but I had success deleting the `bin` and `obj` folders in my MAUI project after I had this issue. Once those were gone, I was able to rebuild successfully.

**Comment by MLB-BIOMETRIC** [comment]
> I don't believe this will help for all situations, but I had success deleting the `bin` and `obj` folders in my MAUI project after I had this issue. Once those were gone, I was able to rebuild successfully.

Sadly doesn't fix my issue.

**Comment by wallym** [comment]
Dele

> I don't believe this will help for all situations, but I had success deleting the `bin` and `obj` folders in my MAUI project after I had this issue. Once those were gone, I was able to rebuild successfully.

Deleting doesn’t fix my problems unfortunately.

**Comment by wallym** [comment]
I'd like to just go back to .NET 7 until a process is in place to resolve this issue.  Does anyone know how to go back and downgrade the version of .NET from 8 to 7 in a maui project?

**Comment by Redth** [comment]
@wallym if you can share the project you had troubles with that would be helpful.  This error should be fixed by referencing the current nuget + installing latest .NET 8, so I suspect something in your setup is amiss (if you tried deleting bin/obj and it still was happening)

**Comment by wallym** [comment]
@Redth saved my bacon and got it resolved.  I'm not convinced that I didn't cause some of my own problem.  :-)

**Comment by luislhg** [comment]
I'm also not able to update my MAUI projects to .NET 8. If I create a new project from scratch it works fine.
I tried
- dotnet workload update (it says there's nothing to update -I have 8.0.100)
- Closing VS
- Deleting obj/bin folders
- Restarting computer

At the beginning there were several errors, but after doing the steps above it always ends up with this one:
![image](https://github.com/dotnet/maui/assets/16278387/4628e28c-b1df-4e6f-b810-8e223a9d10b1)

This happens to every project that I have, in both computers.
All of them I need to create a new project and copy everything over, really bad...

**Comment by Richard-Dufour** [comment]
For me what fixed it was to target ios 17.0 instead of 16.1

<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">17.0</SupportedOSPlatformVersion>
